### PR TITLE
Reenable Mono runtime tests on PRs

### PR DESCRIPTION
**Pull request by akoeplinger**
/cc @steveisok

### CODE REVIEW

File: eng/pipelines/runtime.yml [hunk 1]
```diff
@@ -1078,7 +1078,10 @@ jobs:
...
       runtimeVariant: llvmaot
       condition: >-
-        eq(variables['isFullMatrix'], true)
+        or(
+          eq(dependencies.evaluate_paths.outputs['SetPathVars_libraries.containsChange'], true),
+          eq(dependencies.evaluate_paths.outputs['SetPathVars_mono.containsChange'], true),
+          eq(variables['isFullMatrix'], true))
 
 #
 # Mono CoreCLR runtime Test executions using live libraries and LLVM Full AOT
```
Comment (lambdageek) [inline eng/pipelines/runtime.yml:1087]: Can we do this one too.

and minijit (OSX x64)

File: eng/pipelines/runtime.yml [hunk 2]
```diff
@@ -1078,7 +1078,10 @@ jobs:
       liveRuntimeBuildConfig: release
       runtimeVariant: llvmaot
       condition: >-
-        eq(variables['isFullMatrix'], true)
+        or(
+          eq(dependencies.evaluate_paths.outputs['SetPathVars_libraries.containsChange'], true),
+          eq(dependencies.evaluate_paths.outputs['SetPathVars_mono.containsChange'], true),
```
Comment (lambdageek) [inline eng/pipelines/runtime.yml:1083]: Should also check `SetPathVars_runtimetests.containsChange` I think...

Comment (lambdageek) [inline eng/pipelines/runtime.yml:1083]: We should also run the tests if `SetPathVars_runtimetests.containsChange` is true, I think

File: eng/pipelines/runtime.yml [hunk 3]
```diff
@@ -1078,7 +1078,10 @@ jobs:
       liveRuntimeBuildConfig: release
       runtimeVariant: llvmaot
       condition: >-
-        eq(variables['isFullMatrix'], true)
+        or(
+          eq(dependencies.evaluate_paths.outputs['SetPathVars_libraries.containsChange'], true),
```
Comment (jkotas) [inline eng/pipelines/runtime.yml:1082]: Is there enough value in running runtime tests for libraries changes? The runtime tests should not be testing much of the libraries.

Comment (steveisok) [reply eng/pipelines/runtime.yml:1082]: No, looks like copy/paste mistake from a long time ago to me.

Comment (akoeplinger) [reply eng/pipelines/runtime.yml]: done

**Comment by akoeplinger** [comment]
Updated based on feedback.

**Comment by akoeplinger** [comment]
@lambdageek there's an error on the `Mono llvmfullaot Pri0 Runtime Tests Run Linux x64 release` job, is that one known?

```
  aot-compile: compiling /__w/1/s/artifacts/tests/coreclr/Linux.x64.Release/Tests/Core_Root/Microsoft.Interop.DllImportGenerator.dll; MONO_PATH: /__w/1/s/artifacts/tests/coreclr/Linux.x64.Release/Tests/Core_Root:/__w/1/s/artifacts/tests/coreclr/Linux.x64.Release/Tests/Core_Root
  Failed to load method 0x6000034 from '/__w/1/s/artifacts/tests/coreclr/Linux.x64.Release/Tests/Core_Root/Microsoft.Interop.DllImportGenerator.dll' due to Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies..
  Run with MONO_LOG_LEVEL=debug for more information.
  AOT of image /__w/1/s/artifacts/tests/coreclr/Linux.x64.Release/Tests/Core_Root/Microsoft.Interop.DllImportGenerator.dll failed.
  Mono Ahead of Time compiler - compiling assembly /__w/1/s/artifacts/tests/coreclr/Linux.x64.Release/Tests/Core_Root/Microsoft.Interop.DllImportGenerator.dll
```

**Comment by lambdageek** [comment]
> @lambdageek there's an error on the `Mono llvmfullaot Pri0 Runtime Tests Run Linux x64 release` job, is that one known?
> 
> ```
>   aot-compile: compiling /__w/1/s/artifacts/tests/coreclr/Linux.x64.Release/Tests/Core_Root/Microsoft.Interop.DllImportGenerator.dll; MONO_PATH: /__w/1/s/artifacts/tests/coreclr/Linux.x64.Release/Tests/Core_Root:/__w/1/s/artifacts/tests/coreclr/Linux.x64.Release/Tests/Core_Root
>   Failed to load method 0x6000034 from '/__w/1/s/artifacts/tests/coreclr/Linux.x64.Release/Tests/Core_Root/Microsoft.Interop.DllImportGenerator.dll' due to Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies..
>   Run with MONO_LOG_LEVEL=debug for more information.
>   AOT of image /__w/1/s/artifacts/tests/coreclr/Linux.x64.Release/Tests/Core_Root/Microsoft.Interop.DllImportGenerator.dll failed.
>   Mono Ahead of Time compiler - compiling assembly /__w/1/s/artifacts/tests/coreclr/Linux.x64.Release/Tests/Core_Root/Microsoft.Interop.DllImportGenerator.dll
> ```

I don't think I've seen this before.  I don't think that DllImportGenerator.dll should be copied to Core_Root (or at least it shouldn't be AOTd once it's in Core_Root) - I think it's a C# source generator.

**Comment by steveisok** [comment]
> #62372 ?

Should we skip anything before enabling?

**Review by steveisok — APPROVED** [review]

**Pull request merged**